src/arith/rewrite_simplify.cc

```diff
@@ -1136,8 +1136,15 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const FloorDivNode* op) {
                        x + floordiv(y, z), CanProveGreaterEqual(z.Eval(), 0));
     TVM_TRY_REWRITE_IF(matches_one_of(floordiv(y + x * z, z), floordiv(y + z * x, z)),
                        floordiv(y, z) + x, CanProveGreaterEqual(z.Eval(), 0));
+    TVM_TRY_REWRITE_IF(floordiv(x * z * c1 + y, z * c1), x + floordiv(y, z * c1),
+                       CanProveGreaterEqual(z.Eval() * c1.Eval(), 0));
```

Comment:
If `z * c1 == 0` we get a division by zero? I hope it would not be allowed in constructing the `floordiv`, but maybe we shouldn't explicitly allow it in rewrite rules.

Reply:
Yes, I wasn't sure what to do about it, especially since the earlier rewrite rules don't seem to address this case either. 
I've tried out a few more testcases and at least setting the constant `c1` to 0 does make it fail during `TestCase(fld(x * z * 0 + y, z * 0), x + fld(y, z * 0))` with `InternalError: Check failed: pb->value != 0 (0 vs. 0) : Divide by zero` in `tvm::arith::TryConstFold`, but this similar case doesn't fail although it preferably should: `TestCase(fld(x * z * 2 + y, z * 2), x + fld(y, z * 2), [z >= 0, z <= 0])`.

Since there is no `CanProveGreater()` function, I think we could either:

1. use `CanProveLess(-1 * z.Eval() * c1.Eval(), 0)` which uses the same simple `const_int_bound` check but looks a bit awkward
2. use `CanProve(z.Eval() * c1.Eval() > 0)` which is clear but _might_ be a costlier check
3. write a new `CanProveGreater()` function in `analyzer.cc`

Reply:
I'd lean towards option 3, it's pretty straightforward to implement (it can even just be a wrapper around 1). WDYT? 

To me it is a bit arbitrary that `CanProveLess` and `CanProveGreaterEqual` don't have same powers as `CanProve` and `CanProveEqual`, but I suppose the tradeoffs there are a topic of another discussion :)

Reply:
I went ahead and implemented the `CanProveGreater()` function and took the opportunity to change all other cases of possible division by zero that I could find in the `rewrite_simplify.cc` rules to use it too.